homeassistant/components/viam/__init__.py

```diff
@@ -0,0 +1,330 @@
+            )
+            image = RawImage(cam_bytes, "jpeg") if len(cam_bytes) > 0 else None
+
+        if image is None:
+            return {
+                "classifications": [],
+                "img_src": filepath or None,
+            }
+
+        img_src = filepath or self._encode_image(image)
+        classifications = await classifier.get_classifications(image, count)
+
+        return {
+            "classifications": [
+                {"name": c.class_name, "confidence": c.confidence}
+                for c in classifications
+                if c.confidence >= confidence_threshold
+            ],
+            "img_src": img_src,
+        }
+
+    async def get_detections(self, call: ServiceCall) -> ServiceResponse:
+        """Accept input configuration to request detections."""
+        filepath = call.data.get("filepath")
+        camera = call.data.get("camera")
+        detector_name = call.data.get("detector_name")
+        confidence_threshold = float(call.data.get("confidence_threshold", 0.6))
+
+        async with await self._get_robot_client(
+            call.data.get("robot_secret"), call.data.get("robot_address")
+        ) as robot:
+            detector = VisionClient.from_robot(robot, detector_name)
+            image = Image.open(filepath) if filepath is not None else None
+            cam_bytes = (
+                await self._get_image_from_camera(camera) if camera is not None else b""
+            )
+            image = RawImage(cam_bytes, "jpeg") if len(cam_bytes) > 0 else None
+
+        if image is None:
+            return {
+                "detections": [],
+                "img_src": filepath or None,
+            }
+
+        img_src = filepath or self._encode_image(image)
+        detections = await detector.get_detections(image)
+
+        return {
+            "detections": [
+                {
+                    "name": c.class_name,
+                    "confidence": c.confidence,
+                    "x_min": c.x_min,
+                    "y_min": c.y_min,
+                    "x_max": c.x_max,
+                    "y_max": c.y_max,
+                }
+                for c in detections
+                if c.confidence >= confidence_threshold
+            ],
+            "img_src": img_src,
+        }
+
+    def _encode_image(self, image: RawImage):
+        image_string = (
+            str(base64.b64encode(image.data))
+            .replace("\\n", "")
+            .replace("b'", "")
+            .replace("'", "")
+        )
+        return f"data:image/png;base64,{image_string}"
+
+    async def _get_image_from_camera(self, camera_name: str) -> bytes:
+        cam_entity = er.async_get(self.hass).async_get(camera_name)
+        if cam_entity is None:
+            raise ServiceValidationError(
+                f"A camera entity called {camera_name} was not found",
+                translation_key="camera_entity_not_found",
```

Comment:
```suggestion
                translation_domain=DOMAIN,
                translation_key="camera_entity_not_found",
```

Please adopt the other places too